test/ffmpeg-qsv/transcode/transcoder.py

```diff
@@ -10,42 +10,140 @@
-    else:
-      self.mjpeg_quality = ""
+    assert len(self.outputs), "Invalid test case specification, outputs data empty"
+    assert self.mode in ["sw", "hw"], "Invalid test case specification as mode type not valid"
 
-    get_media().test_call_timeout = vars(self).get("call_timeout", 0)
+    # generate platform list based on test runtime parameters
+    iplats, ireq, _ =  self.get_requirements_data("decode", self.codec, self.mode)
+    platforms = set(iplats)
+    requires = [ireq,]
+
+    for output in self.outputs:
+      codec = output["codec"]
+      mode  = output["mode"]
+      assert mode in ["sw", "hw"], "Invalid test case specification as output mode type not valid"
+      oplats, oreq, _ = self.get_requirements_data("encode", codec, mode)
+      platforms &= set(oplats)
+      requires.append(oreq)
+
+    # create matchers based on command-line filters
+    matchers = [Matcher(s) for s in slash.config.root.run.filter_strings]
+
+    # check if user supplied any platform tag on command line
+    pmatch = any(map(lambda p: any([m.matches(p) for m in matchers]), ALL_PLATFORMS))
+
+    # if user supplied a platform tag, check if this test can support it via the
+    # test param-based required platforms
+    if pmatch and not any(map(lambda p: any([m.matches(p) for m in matchers]), platforms)):
+      slash.skip_test("unsupported platform")
 
-    if vars(self).get("mode", None) == 'hwhw':
-      self.output = call(
-        "ffmpeg -hwaccel qsv -hwaccel_device /dev/dri/renderD128 -v verbose"
-        " -c:v {ffdecoder} -i {source} -an -init_hw_device qsv:hw -c:v {ffencoder}"
-        " -vframes {frames} -y {decoded}".format(**vars(self)))
-    elif  vars(self).get("mode", None) == 'hwsw':
-      self.output = call(
-        "ffmpeg -v verbose -init_hw_device qsv:hw"
-        "  -c:v {ffdecoder} -i {source}  -an -c:v {ffencoder}"
-        " {mjpeg_quality} -vframes {frames} -y {decoded}".format(**vars(self)))
-    elif  vars(self).get("mode", None) == 'swhw':
-      self.output = call(
-        "ffmpeg -hwaccel qsv -init_hw_device qsv=hw -filter_hw_device hw -v verbose "
-        " -c:v {ffdecoder} -i {source} -an -vf hwupload=extra_hw_frames=64,format=qsv  -c:v {ffencoder}"
-        " {mjpeg_quality} -vframes {frames} -y {decoded}".format(**vars(self)))
+    # check required
+    if not all([t for t,m in requires]):
+      slash.skip_test("One or more software requirements not met")
+
+  def gen_input_opts(self):
+    opts = ""
+     
+    _, _, ffparms = self.get_requirements_data("decode", self.codec, self.mode)
+    assert ffparms is not None, "gen_input_opts decode parameters are empty"
+
+    ffcodec, _ = ffparms
+
+    if "hw" == self.mode:
+      opts += " -hwaccel qsv -hwaccel_device /dev/dri/renderD128 -init_hw_device qsv:hw -c:v {} ".format(ffcodec)
     else:
-        assert "non supported transcoding"
+      opts += " -hwaccel qsv -init_hw_device qsv=hw -filter_hw_device hw -c:v {} ".format(ffcodec)
```

Comment:
It looks like you use two different syntax for `-init_hw_device` in "sw" vs. "hw" mode.  I'm not sure if ffmpeg supports both syntax... but we should only use one syntax for both modes.

Reply:
changed to "-init_hw_device qsv=hw "
for hwsw mode, it is taking both types but for swhw mode it is accepting only above syntax. (hwsw mode tests are running without above also)